### PR TITLE
Add MLQuestions dataset

### PR DESCRIPTION
**Pull request by vaibhavad**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by vaibhavad** [comment]
This is ready to be merged. However `make test` is failing but it seems it is not due to any changes in this branch

```python
>           assert False, f"Datasets not available on Hugging Face:\n{pretty_print}"
E           AssertionError: Datasets not available on Hugging Face:
E             iwslt2017 - revision c18a4f81a47ae6fa079fe9d32db288ddde38451d
E             myanmar_news - revision b899ec06227db3679b0fe3c4188a6b48cc0b65eb
E             moroco - revision d64d9b8cd876056a5c24552afe3caf7e6fd26c8e
E             ro_sent - revision 155048684cea7a6d6af1ddbfeb9a04820311ce93
E             swedish_reviews - revision 105ba6b3cb99b9fd64880215be469d60ebf44a1b
E             indonli - revision 3c976110fc13596004dc36279fc4c453ff2c18aa
E             ro_sts - revision 41a33183b739070f3d46d9d446492c1d2f98ce1a
E           assert False

tests/test_all_abstasks.py:76: AssertionError
```

Also `make lint` is changing 100+ files, also not linked to this PR. I am not sure if I should commit those

**Comment by KennethEnevoldsen** [comment]
@vaibhavad I have added a PR #803 which should fix the issues. A pull from main should resolve the issues.

File: mteb/tasks/Retrieval/eng/MLQuestions.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            "It consists of ML questions along with passages from Wikipedia machine learning pages (https://en.wikipedia.org/wiki/Category:Machine_learning)"
+        ),
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["dev", "test"],
+        eval_langs=["eng-Latn"],
+        main_score="ndcg_at_10",
+        date=("2021-01-01", "2021-03-31"),
+        form=["written"],
+        domains=["Encyclopaedic"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/MLQuestions.py:29]:
```suggestion
        domains=["Encyclopaedic", "Academic"],
```
I suppose

File: mteb/tasks/Retrieval/eng/MLQuestions.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        task_subtypes=["Article retrieval"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/MLQuestions.py:30]:
```suggestion
        task_subtypes=["Question Answering"],
```
I believe article retrieval typically refers to news article retrieval, but we might want to rename it to match that.

File: mteb/tasks/Retrieval/eng/MLQuestions.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        description=(
+            "MLQuestions is a domain adaptation dataset for the machine learning domain"
+            "It consists of ML questions along with passages from Wikipedia machine learning pages (https://en.wikipedia.org/wiki/Category:Machine_learning)"
+        ),
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["dev", "test"],
+        eval_langs=["eng-Latn"],
+        main_score="ndcg_at_10",
+        date=("2021-01-01", "2021-03-31"),
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/MLQuestions.py:27]: Is this the period in which the wiki articles were collected? (would add it as a comment)

File: docs/mmteb/points/799.jsonl [hunk 4]
```diff
@@ -0,0 +1 @@
+{"GitHub": "vaibhavad", "New dataset": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/799.jsonl:1]:
```suggestion
{"GitHub": "vaibhavad", "New dataset": 2}
{"GitHub": "KennethEnevoldsen", "PR Review": 2}
```

**Review by KennethEnevoldsen — APPROVED** [review]
A few minor comments otherwise I believe it looks good!

File: mteb/tasks/Retrieval/eng/MLQuestions.py (hunk 3 above)
Comment (vaibhavad) [reply mteb/tasks/Retrieval/eng/MLQuestions.py:27]: Yes both wiki articles and queries, although the original publish date of those articles will vary widely

**Comment by vaibhavad** [comment]
@KennethEnevoldsen, I added all the suggestions, I'll merge it now. Thanks for reviewing and providing feedback!

**Pull request merged**